src/gui/title_bar.cpp

```diff
@@ -0,0 +1,600 @@
+// Lifecycle
+// ***************************************************************************
+
+void TITLEBAR_ReadConfig(Section* section)
+{
+	assert(section);
+	const Section_prop* conf = dynamic_cast<Section_prop*>(section);
+	assert(conf);
+	if (!conf) {
+		return;
+	}
+
+	bool warned_double_animation = false;
+	bool warned_double_cycles    = false;
+	bool warned_double_dosbox    = false;
+	bool warned_double_program   = false;
+	bool warned_double_version   = false;
+
+	auto check_not_set_animation = [&]() {
+		if (config.animated_rec && !warned_double_animation) {
+			LOG_WARNING("SDL: 'animation' specified more than once in 'titlebar_content'");
+			warned_double_animation = true;
+		}
+	};
+
+	auto check_not_set_cycles = [&]() {
+		if (config.show_cycles && !warned_double_cycles) {
+			LOG_WARNING("SDL: 'cycles' specified more than once in 'titlebar_content'");
+			warned_double_cycles = true;
+		}
+	};
+
+	auto check_not_set_dosbox = [&]() {
+		if (config.show_dosbox && !warned_double_dosbox) {
+			LOG_WARNING("SDL: 'dosbox' specified more than once in 'titlebar_content'");
+			warned_double_dosbox = true;
+		}
+	};
+
+	auto check_not_set_program = [&]() {
+		if (config.program != ProgramDisplay::None && !warned_double_program) {
+			LOG_WARNING("SDL: 'program' specified more than once in 'titlebar_content'");
+			warned_double_program = true;
+		}
+	};
+
+	auto check_not_set_version = [&]() {
+		if (config.version != VersionDisplay::None && !warned_double_version) {
+			LOG_WARNING("SDL: 'version' specified more than once in 'titlebar_content'");
+			warned_double_version = true;
+		}
+	};
+
+	config = TitlebarConfig();
+
+	const auto titlebar_content = split(conf->Get_string("titlebar_content"));
+	for (const auto& setting : titlebar_content) {
+		if (iequals(setting, "animation")) {
+			check_not_set_animation();
+			config.animated_rec = true;
+			continue;
+		}
+
+		if (iequals(setting, "cycles")) {
+			check_not_set_cycles();
+			config.show_cycles = true;
+			continue;
+		}
+
+		if (iequals(setting, "dosbox")) {
+			check_not_set_dosbox();
+			config.show_dosbox = true;
+			continue;
+		}
+
+		if (iequals(setting, "program") ||
+		    iequals(setting, "program=name")) {
+			check_not_set_program();
+			config.program = ProgramDisplay::Name;
+			continue;
+		}
+
+		if (iequals(setting, "program=path")) {
+			check_not_set_program();
+			config.program = ProgramDisplay::Path;
+			continue;
+		}
+
+		if (iequals(setting, "program=segment")) {
+			check_not_set_program();
+			config.program = ProgramDisplay::Segment;
+			continue;
+		}
+
+		if (iequals(setting, "version") ||
+		    iequals(setting, "version=simple")) {
+			check_not_set_version();
+			config.version = VersionDisplay::Simple;
+			continue;
+		}
+
+		if (iequals(setting, "version=detailed")) {
+			check_not_set_version();
+			config.version = VersionDisplay::Detailed;
+			continue;
+		}
+
+		LOG_WARNING("SDL: Unknown setting '%s' in 'titlebar_content'",
```

Comment:
The new consistent pattern I introduced a while ago (I reworked most if not all such error messages) would be this:

  `SDL: Invalid 'titlebar_content' setting '<setting-value>', using
  '<the-default-value'`

  Do a `git grep ', using ` for tons of examples.

Reply:
When you change a config property programmatically (e.g., when defaulting because the user provided an invalid setting), you should also sync the config property to the changed setting. Otherwise `config -h <settingname>` would still display the invalid value!

This will be a major problem for the OSD too. I've started doing such syncing in my new code; check out `sync_crossfeed_setting` and `sync_reverb_setting` in `mixer.cpp`, or search for `SetValue` in `render.cpp`.

Reply:
OK, changed warning messages and added the synchronization code.